app/modules/search/controllers/searchControllers.js

```diff
@@ -126,22 +303,34 @@
                     }
                 }
             }
-            ;
-            //Perform search through client and get a search Promise
-            searchclient.search({
+            var request = {
                 index: API_CONF.ELASTIC_INDEX_NAME,
                 type: $scope.searchItemType,
                 body: {
                     size: $scope.itemsperPage,
                     from: itemOffset,
                     sort: sort,
-                    query: query
+                    query: query,
+                    aggs: requestAggs
                 }
-            }).then(function (resp) {
+            };
+            var filters = normalizeAggregationFilter();
+            if (!angular.equals({}, filters)) {
+                request.body.post_filter = filters;
+                //request.body.query = {
```

Comment:
@slomo are you referring to different line(s) than the 316-328 block? The commented block here, exists in perpose since it's changing the faceting population and we might need it when the current behaviour is evaluated.